File: app/src/main/java/com/github/android/lvrn/lvrnproject/persistent/entity/TrashDependedEntity.java
package com.github.android.lvrn.lvrnproject.persistent.entity;

import android.support.annotation.NonNull;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public abstract class TrashDependedEntity extends ProfileDependedEntity {

    /**
     * A field which shows trash status of an entity.
     */
    protected boolean isTrash;

    public boolean isTrash() {
        return isTrash;
    }

    public void setTrash(boolean trash) {
        isTrash = trash;
    }

    @NonNull
    @Override
    public String toString() {
        return "TrashDependedEntity{" + super.toString() +
                "isTrash=" + isTrash +
                '}';
    }
}

File: app/src/main/java/com/github/valhallalabs/laverna/persistent/entity/base/TrashDependedEntity.kt
package com.github.valhallalabs.laverna.persistent.entity.base

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
 */
abstract class TrashDependedEntity : ProfileDependedEntity() {
    /**
     * A field which shows trash status of an entity.
     */
    abstract val isTrash: Boolean

    override fun toString(): String {
        return "TrashDependedEntity{" + super.toString() +
                "isTrash=" + isTrash +
                "}"
    }
}